test/Benchmarks/Internal/CircularBufferBenchmarks.cs

```diff
@@ -0,0 +1,92 @@
+// <copyright file="CircularBufferBenchmarks.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Benchmarks.Internal
+{
+    public class CircularBufferBenchmarks
+    {
+        [Benchmark]
+        public void TryAddBenchmark()
```

Comment:
We should leverage the stress test infra : https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/test/OpenTelemetry.Tests.Stress

Its been used extensively while working on Metrics SDK.

(btw, the same issue about yielding - is applicable in metric aggregators as well. need to fix them as well based on the outcome of this.)

Reply:
@cijothomas I made a simple stress test here - https://github.com/open-telemetry/opentelemetry-dotnet/pull/1424/commits/177c9bf107c518e57986c48060b757c42d64e1c3, is it close to what you want? I'm sorry if not, it's really hard to get back to that code year later :)